R/utils.R

```diff
@@ -645,3 +645,27 @@ mod_first <- function(cl) {
   msg <- sprintf("the first argument to %s must be a model object",fun)
   wstop(msg)
 }
+
+#' Parse a file name
+#' @param x a file name to parse.
+#' @return 
+#' A list with `ext`, `root` and `has_ext`.
+#' @details
+#' This follows the approach used in [tools::file_path_sans_ext].
```

Comment:
Probably fine in this context, but `tools::file_path_sans_ext` is a bit stricter regarding file names starting with a dot:

``` r
tools::file_path_sans_ext(".foo")
#> [1] ".foo"
```

It looks like only the `root` value is used in this PR, so I'm curious why it's worth adding a custom function instead of using `tools::file_path_sans_ext` (but the code looks fine to me, aside from perhaps the ".foo" behavior).

Reply:
The motivation was to avoid making two passes through the string since I wanted the extension as well. `file_ext()` will get the position of the `.` on the extension and if we could get access to that, it gives clean way to split the string. So I wrote that myself and then used the regmatch to split. I can change it to boxed code.

Reply:
This is kind of weird, though:

```
> tools:::file_path_sans_ext(".foo")
[1] ".foo"
> tools:::file_ext(".foo")
[1] "foo"
```

Reply:
> The motivation was to avoid making two passes through the string since I wanted the extension as well.

Right, my confusion is that I'm not seeing where the extension is ever used (perhaps I'm missing that, or perhaps you want it because you expect to use it for some future feature?)

> I can change it to boxed code.

If you prefer the custom function, no objections from me.  I just wasn't sure of the motivation.

> This is kind of weird, though:

Hmm, yeah, that inconsistency is weird.  

I guess it doesn't really matter in this context, given a cppfile with a leading dot would be unlikely, but it seems like the `file_path_sans_ext` behavior is probably more appropriate, right?

Reply:
Ok; dropped the custom code and no longer tracking `ext`.  Dropped the test too. Will re-visit if / when I need to track `ext`. 

Thanks.